File: common/src/main/java/net/hexdummy/HexDummy.java
package net.hexdummy;

import net.hexdummy.registry.HexDummyIotaTypeRegistry;
import net.hexdummy.registry.HexDummyItemRegistry;
import net.hexdummy.registry.HexDummyPatternRegistry;
import net.hexdummy.networking.HexDummyNetworking;
import net.minecraft.resources.ResourceLocation;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * This is effectively the loading entrypoint for most of your code, at least
 * if you are using Architectury as intended.
 */
public class HexDummy {
    public static final String MOD_ID = "hexdummy";
    public static final Logger LOGGER = LogManager.getLogger(MOD_ID);


    public static void init() {
        LOGGER.info("Hex Dummy says hello!");

        HexDummyAbstractions.initPlatformSpecific();
        HexDummyItemRegistry.init();
        HexDummyIotaTypeRegistry.init();
        HexDummyPatternRegistry.init();
		HexDummyNetworking.init();

        LOGGER.info(HexDummyAbstractions.getConfigDirectory().toAbsolutePath().normalize().toString());
    }

    /**
     * Shortcut for identifiers specific to this mod.
     */
    public static ResourceLocation id(String string) {
        return new ResourceLocation(MOD_ID, string);
    }
}

File: common/src/main/kotlin/io/github/fallingcolors/hexdummy/HexDummy.kt
package io.github.fallingcolors.hexdummy

import net.minecraft.resources.ResourceLocation
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import io.github.fallingcolors.hexdummy.config.HexDummyServerConfig
import io.github.fallingcolors.hexdummy.networking.HexDummyNetworking
import io.github.fallingcolors.hexdummy.registry.HexDummyActions

object HexDummy {
    const val MODID = "hexdummy"

    @JvmField
    val LOGGER: Logger = LogManager.getLogger(MODID)

    @JvmStatic
    fun id(path: String) = ResourceLocation(MODID, path)

    fun init() {
        HexDummyServerConfig.init()
        initRegistries(
            HexDummyActions,
        )
        HexDummyNetworking.init()
    }

    fun initServer() {
        HexDummyServerConfig.initServer()
    }
}
